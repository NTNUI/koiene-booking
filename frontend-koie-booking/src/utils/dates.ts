import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
