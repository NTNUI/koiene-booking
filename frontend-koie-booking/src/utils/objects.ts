export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
