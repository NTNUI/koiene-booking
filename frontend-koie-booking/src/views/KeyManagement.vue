<template>
  <KeyManagerOverview style="padding: 10px" />
</template>

<script lang="ts">
import Vue from 'vue';
import KeyManagerOverview from '@/components/keyManager/KeyManagerOverview.vue';

export default Vue.extend({
  name: 'KeyManagement',
  components: { KeyManagerOverview },
});
</script>

<style scoped></style>
