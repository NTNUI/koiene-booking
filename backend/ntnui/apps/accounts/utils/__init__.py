from accounts.utils.mail_utils import send_gsuite_confirmation_mail
