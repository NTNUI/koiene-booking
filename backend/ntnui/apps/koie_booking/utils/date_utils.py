from django.utils.timezone import timedelta


def get_daterange(start_date, end_date):
    for n in range(int((end_date - start_date).days)):
        yield start_date + timedelta(n)
