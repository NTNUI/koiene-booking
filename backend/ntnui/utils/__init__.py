from ntnui.utils.pagination import PageSize7, PageSize20, PageSize100
