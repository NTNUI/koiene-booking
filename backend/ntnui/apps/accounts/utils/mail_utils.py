from django.core.mail import send_mail
from django.template.loader import render_to_string

from ntnui.enums import MembershipType


def new_membership(membership):
    """ Sends confirmation email to users and guests who signs up for events and sub-events. """

    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [membership.member.email]
    subject = "Welcome"

    context = {"user": membership.member, "group": membership.group}

    # Mail body.
    plain_message = render_to_string("emails/new_membership.txt", context)
    html_message = render_to_string("emails/new_membership.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_gsuite_confirmation_mail(user, contact_email, password):
    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = "Opprettelse av ny NTNUI-bruker"

    context = {
        "first_name": user.first_name,
        "email": user.email,
        "contact_email": contact_email,
        "password": password,
    }

    # Mail body.
    plain_message = render_to_string("emails/new_gsuite_account.txt", context)
    html_message = render_to_string("emails/new_gsuite_account.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_welcome_mail_to_leader(user, group):
    """ Send a mail to a new leader """
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny leder i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_leader_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_leader_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_cashier(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny kasserer i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_cashier_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_cashier_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_board_member(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny tillitsvalgt i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_board_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_board_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail(membership):
    """ Delegate welcome mail to different board members """
    send_welcome_mail_to_board_member(user=membership.member, group=membership.group)
    if membership.type == MembershipType.leader:
        send_welcome_mail_to_leader(user=membership.member, group=membership.group)

    if membership.type == MembershipType.cashier:
        send_welcome_mail_to_cashier(user=membership.member, group=membership.group)
