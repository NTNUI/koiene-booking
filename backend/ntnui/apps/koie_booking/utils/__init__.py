from koie_booking.utils.mail_utils import send_confirmation_mail, send_koie_information_mail
from koie_booking.utils.date_utils import get_daterange
