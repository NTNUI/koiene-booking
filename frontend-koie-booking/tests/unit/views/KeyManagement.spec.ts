import Vue from 'vue';
import Vuetify from 'vuetify';
import { createShallowWrapper } from '../utils';
import KeyManagement from '@/views/KeyManagement.vue';

Vue.use(Vuetify);

describe('View KeyManagement.vue', () => {
  it('Matches snapshot', () => {
    const wrapper = createShallowWrapper(KeyManagement);
    expect(wrapper).toMatchSnapshot();
  });
});
