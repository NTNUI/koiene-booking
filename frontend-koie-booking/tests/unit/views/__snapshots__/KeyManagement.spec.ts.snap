// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View KeyManagement.vue Matches snapshot 1`] = `<keymanageroverview-stub style="padding: 10px;"></keymanageroverview-stub>`;
